sql/core/src/main/java/org/apache/spark/sql/sources/v2/writer/DataSourceWriter.java

```diff
@@ -62,6 +62,15 @@
    */
   DataWriterFactory<Row> createWriterFactory();
 
+  /**
+   * Handles a commit message on receiving from a successful data writer.
+   *
+   * If this method fails (by throwing an exception), this writing job is considered to to have been
+   * failed, and {@link #abort(WriterCommitMessage[])} would be called. The state of the destination
+   * is undefined and {@link #abort(WriterCommitMessage[])} may not be able to deal with it.
```

Comment:
What does it mean that "the state of the destination is undefined"? I think it is sufficient to say that `abort` will be called and the contract for aborting commits applies.

Reply:
makes sense, let's remove the last sentence.